straxen/plugins/peaklet_processing.py

```diff
@@ -146,6 +155,15 @@ def compute(self, records, start, end):
             min_area=self.config['peak_split_min_area'],
             do_iterations=self.config['peak_split_iterations'])
 
+        # Saturation correction using non-saturated channels
+        # similar method used in pax
+        # see https://github.com/XENON1T/pax/pull/712
+        if self.config['saturation_correction_on']:
+            corrected_peaklet_list = peak_saturation_correction(
```

Comment:
I am a bit confused, it looks like that you are modifying the peaks in place, also the return is not used anywhere. So why do you define it in the first place? Will this not just occupy memory?

Reply:
This should indeed go in the compute function

Reply:
The return is a relic from when the code wasn't optimized. (When I was doing sum waveform outside the correction function). Will just delete the return